controllers/common.go

```diff
@@ -13,12 +34,404 @@ func getServiceAccountName() string {
 	return "search-serviceaccount"
 }
 
-func getImagePullSecret() string {
+func getImagePullSecretName() string {
 	return "search-pull-secret"
 }
+
```

Comment:
We could have a common function to get the deployment from name like this:
```go
func getDeploymentConfig(name string, instance *searchv1alpha1.Search) DeploymentConfig {
    switch name {
    case apiDeploymentName:
        return instance.Spec.Deployments.API
    case collectorDeploymentName:
        return instance.Spec.Deployments.Collector
    case indexerDeploymentName:
        return instance.Spec.Deployments.Indexer
    case postgresDeploymentName:
        return instance.Spec.Deployments.Database
    }
    return nil
} 
```

Then the functions below can be simplified like this:
```go
func getNodeSelector(deploymentName string, instance *searchv1alpha1.Search) map[string]string {
        deploymentConfig := getDeploymentConfig(deploymentName, instance)
        if deploymentConfig.NodeSelector != nil {
            return deploymentConfig.NodeSelector
        }
	var result map[string]string
        return result
}
```

Reply:
Fixed it